docs/javascript/v2/how-to-guides/enable-content-sharing/pdf-annotations.mdx

```diff
@@ -0,0 +1,196 @@
+## Building custom UI
+If you aren't using our Sample App and are building a completely custom UI, there are a few steps required to enable PDF annotations and sharing.
+
+<StepsToc
+  parentId="pdf-sharing"
+  descriptions={[
+    "Add UI elements to enable PDF sharing for your users and enhance collaboration.",
+    "Add UI to prompt the user to select a local PDF file or provide a URL.",
+    "Add an iframe to embed the PDF viewer within your UI to render the selected PDF and start sharing.",
+  ]}
+/>
+
+<StepsContainer id="pdf-sharing">
+
+### Add UI to Enable PDF Sharing
+
+![Enable PDF Sharing](/docs/v2/pdf-sharing/pdf-sharing-start-sharing.png)
+
+Add UI elements to enable PDF sharing for your users. Use role permissions to decide if you want to allow PDF sharing for users in a room. Note: Disable PDF sharing UI for users when there's an ongoing sharing session. 
+
+You can refer to the code snippet in our repository to learn more about how we built this. [View Snippet](https://github.com/100mslive/100ms-web/blob/main/src/components/pdfAnnotator/shareScreenOptions.jsx)
+
+### Add UI to Enable PDF Selection
+
+![PDF Selection](/docs/v2/pdf-sharing/pdf-sharing-user-input.png)
+
+Next, you need to prompt the user to provide input, either by selecting a local PDF file or providing a PDF URL. We also recommend implementing validation to handle scenarios such as:
+  1. Invalid local PDF file.
+  2. Invalid or non-PDF URL.
+
+Check out the code snippet in our repository to see how we built this. [View Snippet](https://github.com/100mslive/100ms-web/blob/main/src/components/pdfAnnotator/pdfFileOptions.jsx)
+
+### Render and Share the PDF
+
+![Embed PDF Viewer](/docs/v2/pdf-sharing/pdf-sharing-embed.png)
+
+Finally, add an iframe element that points to the PDF viewer and start screen sharing. You can refer to the code snippet and comments below:
+
+```jsx
+import { useCallback, useEffect, useRef, useState } from "react";
+import { throwErrorHandler, useScreenShare } from "@100mslive/react-sdk";
+import { Box, ThemeTypes, useTheme } from "@100mslive/react-ui";
+
+/**
+ * PDFEmbedComponent is responsible for rendering the PDF iframe and managing the screen sharing functionality.
+ * It receives the pdfConfig and setPDFConfig as props for accessing and updating the PDF configuration state.
+ * 
+ * pdfConfig ({isPdfBeingShared: boolean})
+ */
+export const PDFEmbedComponent = ({ pdfConfig, setPDFConfig }) => {
+  let pdfIframeURL = process.env.REACT_APP_PDFJS_IFRAME_URL;
+
+  // Check if the current browser is Chrome to force current tab selection
+  const isChrome = parsedUserAgent.getBrowser()?.name?.toLowerCase() === "chrome";
+  const pdfIframeRef = useRef(); // Create a ref to access the iframe element
+  const themeType = useTheme().themeType; // Get the current theme type from the theme context
+
+  const [isPDFLoaded, setIsPDFLoaded] = useState(false);
+  const { amIScreenSharing, toggleScreenShare } =
+    useScreenShare(throwErrorHandler);
```

Comment:
this is not a right abstraction ... if someone needs to pass `throwErrorHandler` .. why cant we just abstract it ?